### ISSUE TITLE
[HELP] How to link LL on react-native 0.61

### ISSUE THREAD

**Issue by chgsilva** [localytics/react-native-template-app#18]
Is there any kind of guide on how to auto link under React native version 0.61+? 

Since manual link is not supported anymore, I don't know how to properly link locallytics.

**Comment by bradenpowers** [comment]
I am having the same issues trying to get this to work. I am on 0.61.5.

**Comment by Mriegel** [comment]
@chgsilva, as of right now the Localytics React-Native wrapper does not support autolinking.  We are working to improve upon this and are hoping to have autolinking resolved and fixed in our next release.

As for workarounds right now - there is a way to make the current module work within RN 0.61.x. While auotlinking is the preferred mechanism, manual linking can still work.  In order to get this running, you should follow the integration steps listed on our [docs site](https://docs.localytics.com/dev/react-native.html#ios-integration-react-native) with a few modifications.  

Once those have been followed you can then fix one underlying issue. This issue stems from XCode's inability to find the React-Native headers for projects that don't use autolinking.  We can fix that by specifying where those live. The errors I saw were along the lines of: `React/RCTEventEmiiter.h file not found`

These can be resolved by pointing the HEADER_SEARCH_PATHS under build settings to the right path. Select the “Build Settings” tab of the LLLocalytics.xcodeproj project in the Xcode project navigator, and add the following to Header Search Paths: 

![image](https://user-images.githubusercontent.com/1657263/73294263-7c726c80-41d3-11ea-8a02-fadcfd36f216.png)

The paths to add look like this on my project:

- "$(SRCROOT)/../../../ios/Pods/Headers/Public/React-Core" (Recursive)
- "$(SRCROOT)/../../react-native/ReactCommon/yoga" . (Non-Recursive)

To ensure the correctness of those paths, I actually found the paths in finder and dragged them into the Header Search Paths dialog. You should be looking for:

1. React-Core within `iOS/Pods/Headers/Public`
2. yoga within `node_modules/react-native/ReactCommon/`

This should resolve all build issues with the Localytics RN wrapper.  There is one major downside here - the changes being made are only made in the `node_modules`, and therefore not persisted across dependency updates, as well as not checked into version control.  While it is obviously not ideal, it is the solution for the time being until we can get auto-linking supported. 

To my understanding, autolinking works fine on Android.

**Comment by bradenpowers** [comment]
I will give this a try

**Comment by chgsilva** [comment]
Hello @Mriegel 

I managed to create a custom podspec and I got it working with auto link. What do you think it's better?
You only have to add the summary on the package json, put this file in the node_modules and then link using Podfile `  pod 'localytics-react-native', :path => '../node_modules/localytics-react-native'`
Here is my custom podscpec if anyone's interested: 
```
require 'json'

package = JSON.parse(File.read(File.join(__dir__, 'package.json')))

Pod::Spec.new do |s|
  s.name           = package['name']
  s.version        = package['version']
  s.description    = package['description']
  s.summary        = package['summary']
  s.homepage        = package['homepage']
  s.author         = package['author']
  s.license        = package['license']
  s.source         = { :git => 'https://github.com/localytics/react-native-template-app.git', :tag => "v#{s.version}" }
  s.requires_arc   = true
  s.platform       = :ios, '8.0'
  s.source_files   = 'ios/*.{h,m}', 'ios/PluginLibrary/*.{h,m}'
  s.ios.vendored_frameworks = 'ios/Frameworks/Localytics.framework'
  s.dependency     'React'
end

```

**Comment by Mriegel** [comment]
The podspec looks sound. I'll pull it into the project and do some testing with it over the next couple of days. Will report back with any findings at that point.

**Comment by chgsilva** [comment]
nice, let me know your findings

**Comment by Mriegel** [comment]
@chgsilva So far so good. With the use of the Pod you provided I have updated my project locally and everything seems to be working as intended.  I will include this in the next release of our SDK.

I am still testing to ensure autolinking is working 100% on Android, but once I have all of that resolved I will cut a new release, and let you know when it is availiable

**Comment by Mriegel** [comment]
I have just released version 2.8.2 that should properly support autolinking.  Let me know if you have any issues with it. 

Seeing as this has been resolved I will close the issue - but feel free to reach out again if you have any further issue. And thanks for all your help!
